lib/serve.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+const path = require('path')
+
+const Boom = require('boom')
+const Hapi = require('hapi')
+const pify = require('pify')
+
+function executeAPI (handler, request) {
+  return Promise.resolve()
+    .then(() => handler(request))
+}
+
+function getAPI (cwd, name) {
+  const apiPath = path.join(cwd, 'api', name, 'index.js')
+  let api
+  try {
+    api = require(apiPath)
+  } catch (err) {
+    // do nothing
+  }
+  // property names in require.cache are absolute paths
+  delete require.cache[apiPath]
+  return api
+}
+
+function simplifyRequest (request) {
+  return {
+    body: request.payload,
+    headers: request.headers,
+    host: request.info.host,
+    hostname: request.info.hostname,
+    method: request.method,
+    params: request.params,
+    query: request.query,
+    url: request.url
+  }
+}
+
+function startServer (options) {
+  options = options || {}
+  const server = new Hapi.Server()
+  server.connection({ port: options.port })
+
+  server.route({
+    config: {
+      cors: true // development only, not recommended in production
+    },
+    handler (request, reply) {
+      const api = getAPI(options.cwd, request.params.api)
```

Comment:
cool, i figured hapi probably did something like that. its would be so annoying if you had to code that stuff up yourself for every project using hapi